sql/src/test/java/org/apache/druid/sql/calcite/util/SqlTestFramework.java

```diff
@@ -852,11 +862,16 @@ public PlannerFixture(
           new DruidHookDispatcher()
       );
       componentSupplier.finalizePlanner(this);
-      this.statementFactory = QueryFrameworkUtils.createSqlStatementFactory(
+      final SqlToolbox toolbox = new SqlToolbox(
           framework.engine,
           plannerFactory,
-          authConfig
+          new ServiceEmitter("dummy", "dummy", new NoopEmitter()),
+          new NoopRequestLogger(),
+          QueryStackTests.DEFAULT_NOOP_SCHEDULER,
+          new DefaultQueryConfig(ImmutableMap.of()),
+          new SqlLifecycleManager()
       );
+      this.statementFactory = new TestMultiStatementFactory(toolbox, framework.engine, plannerFactory);
```

Comment:
this seems like a hack to support this in `Calcite*Test` classes
I wonder if it would be possible to find a different way to somehow instruct the production classes to enable the processing of these things?

maybe there could be some `PlannerConfig` knob which could control it in which environments its available?

Reply:
I can't really think of a use case outside of testing, and adding it to the config would make it available in production too i think, which I can't think of any legit uses there since I don't think we want to allow multi-statement jdbc statements in the real world. What do you have in mind?